nullable_type.go

```diff
@@ -0,0 +1,92 @@
+package discordgo
```

Comment:
In DiscordGo we try not to include any utilities (except absolutely necessary ones). So I think it would be best to just move this functionality just to a `MarshalJSON` on the `GuildScheduledEventParams`.
Additionally, this could be separated into a different library.